tests/test_cbt.py

```diff
@@ -397,28 +397,19 @@ def test_snapshot_success_with_parent(self, context, mock_lock,
     @mock.patch('VDI.VDI._cbt_op', autospec=True)
     @mock.patch('VDI.VDI._cbt_log_exists', autospec=True)
     @mock.patch('VDI.VDI._ensure_cbt_space', autospec=True)
-    @mock.patch('util.SMlog', autospec=True)
-    def test_snapshot_out_of_space_failure(self, mock_smlog,
-            mock_ensure_space, mock_logcheck, mock_cbt):
+    def test_snapshot_out_of_space_failure(self, mock_ensure_space, mock_logcheck, mock_cbt):
         # Create the test object
         self.vdi = TestVDI(self.sr, self.vdi_uuid)
         # Set initial state
         self._set_initial_state(self.vdi, True)
         snap_uuid = uuid.uuid4()
 
-        # Write util.SMlog calls to smlog_out
-        self.fakesmlog = ""
```

Comment:
Can this not still be done? It seems to be checking an important side-effect of the method call, i.e. that a log message is emitted to say there is insufficient space.

Reply:
The log message was a side-effect of creating a XenError object, which seemed like the wrong place to be logging from, and with the compatibility layers ended up creating a circular import failure because ```xs_errors``` imported ```util```, which imported ```xs_errors```...

I could take a look at fixing that by adding the logging to ```drivers/VDI.py``` where the exception is handled, or alternatively it would be fixable by reinstating the logging side-effect from ```xs_errors.XenError()``` and moving all the logging code out of ```util``` and into a new library, which is a bigger change.

Reply:
OK, that sounds fair.